Makefile

```
@@ -0,0 +1,175 @@
+# Pre-upload check
+.PHONY: pre-upload
+pre-upload:
+	@echo "$(YELLOW)Running pre-upload checks for $(CONTRACT)...$(NC)"
+	@$(MAKE) check-contract CONTRACT=$(CONTRACT)
+	@$(MAKE) build-contract CONTRACT=$(CONTRACT)
+	@python3 scripts/compare_metadata.py $(CONTRACT)
+	@echo "$(GREEN)Pre-upload checks passed!$(NC)"
+
+# Upload contract with history tracking
+.PHONY: upload
+upload:
```

checkmake (0.2.2)

[warning] Target body for "upload" exceeds allowed length of 5 (17).

(maxbodylength)